engine/src/main/java/org/hibernate/search/engine/search/aggregation/spi/CountAggregationBuilder.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.engine.search.aggregation.spi;
+
+public interface CountAggregationBuilder {
+
+	interface TypeSelector {
+		CountValuesAggregationBuilder values();
+
+		CountDocumentAggregationBuilder documents();
+	}
```

Comment:
FWIW with two separate threads, you could just have two builder types and avoid this.